### PR TITLE
Add DB.db method with Try1

### PR DESCRIPTION
**Pull request by gliptak**
https://github.com/functionaljava/functionaljava/issues/266

### CODE REVIEW

**Comment by gliptak** [comment]
I only changed one of the call to ```db``` to exercise the other code path ...

File: core/src/test/java/fj/control/db/TestDbState.java [hunk 1]
```diff
@@ -16,20 +17,18 @@ public void testWriter() throws SQLException {
...
-                Statement s = null;
-                try {
-                    s = c.createStatement();
-                    assertThat(s.executeUpdate("CREATE TABLE TEST(ID INT PRIMARY KEY, NAME VARCHAR(255))"), is(0));
-                    assertThat(s.executeUpdate("INSERT INTO TEST (ID, NAME) VALUES (" + TEN + ", 'FOO')"), is(1));
-                } finally {
-                    DbUtils.closeQuietly(s);
-                }
-                return Unit.unit();
+        DB<Unit> setup = DB.db((Try1<Connection, Unit, SQLException>) t -> {
```
Comment (mperry) [inline core/src/test/java/fj/control/db/TestDbState.java]: c instead of t here for Connection?

Comment (gliptak) [inline core/src/test/java/fj/control/db/TestDbState.java]: updated

File: core/src/test/java/fj/control/db/TestDbState.java [hunk 2]
```diff
@@ -16,20 +17,18 @@ public void testWriter() throws SQLException {
...
+            } finally {
+                DbUtils.closeQuietly(s);
             }
-        };
+            return Unit.unit();
+        });
+
         DB<Option<Integer>> query = new DB<Option<Integer>>() {
             @Override
             public Option<Integer> run(Connection c) throws SQLException {
```
Comment (mperry) [inline core/src/test/java/fj/control/db/TestDbState.java]: @gliptak You might want to do this method here while you are at it.  I will merge and you can raise another PR if you make the change.

Comment (gliptak) [inline core/src/test/java/fj/control/db/TestDbState.java]: @mperry I specifically left this unchanged, as it takes a different path in the codebase (as additional coverage) ...

**Pull request merged**